compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -682,6 +682,7 @@ fn link_dwarf_object<'a>(
         }
 
         // Input rlibs contain .o/.dwo files from dependencies.
+        #[allow(rustc::potential_query_instability)]
         let input_rlibs = cg_results
             .crate_info
             .used_crate_source
```

Comment:
Can you turn the `used_crate_source` field into an `UnordMap`? Then you should be able to do this:
```rust
// Input rlibs contain .o/.dwo files from dependencies.
let input_rlibs = cg_results
    .crate_info
    .used_crate_source
    .items()
    .filter_map(|(_, csource)| csource.rlib.as_ref())
    .map(|(path, _)| path)
    .into_sorted_stable_ord();
```